src/go/vtunnel/test/e2e/connectivity_test.go

```diff
@@ -174,19 +233,102 @@ func TestConnect(t *testing.T) {
 	}, tryInterval, peerCmdTimeout)
 	require.NoError(t, err, "failed to confirm vtunnel peer process is running")
 
-	t.Logf("Sending a request to vtunnel peer process in [%v] over: 127.0.0.1:%v", wslDistroName, peerTCPPort)
-	peerAddr1 := fmt.Sprintf("127.0.0.1:%v", peerTCPPort)
+	t.Logf("Sending a request to vtunnel peer process in [%v] over: 127.0.0.1:%v", wslDistroName, peerTCPPortT)
+	peerAddr1 := fmt.Sprintf("127.0.0.1:%v", peerTCPPortT)
 	out, err := cmdRunWithOutput("wsl", "--distribution", wslDistroName, "--exec", "curl", "--verbose", "--fail-with-body", peerAddr1)
 	require.NoError(t, err, "Failed sending request to vtunnel peer process")
 	require.Contains(t, out, "vtunnel host 1 called.")
 
-	t.Logf("Sending a request to vtunnel peer process in [%v] over: 127.0.0.1:%v", wslDistroName, peerTCPPort2)
-	peerAddr2 := fmt.Sprintf("127.0.0.1:%v", peerTCPPort2)
+	t.Logf("Sending a request to vtunnel peer process in [%v] over: 127.0.0.1:%v", wslDistroName, peerTCPPortT2)
+	peerAddr2 := fmt.Sprintf("127.0.0.1:%v", peerTCPPortT2)
 	out, err = cmdRunWithOutput("wsl", "--distribution", wslDistroName, "--exec", "curl", "--verbose", "--fail-with-body", peerAddr2)
 	require.NoError(t, err, "Failed sending request to vtunnel peer process")
 	require.Contains(t, out, "vtunnel host 2 called.")
 }
 
+func TestMain(m *testing.M) {
+	var err error
+	tmpDir, err = os.MkdirTemp("", "vtunnel-e2e-test")
+	requireNoErrorf(err, "Failed to create a temp directory")
+	defer os.RemoveAll(tmpDir)
+
+	configPath = filepath.Join(tmpDir, configFile)
+
+	logrus.Info("Building vtunnel host binary")
+	err = buildBinaries("../../main.go", "windows", tmpDir)
+	requireNoErrorf(err, "Failed building vtunnel.exe")
+
+	logrus.Info("Building vtunnel peer binary")
+	err = buildBinaries("../../main.go", "linux", tmpDir)
+	requireNoErrorf(err, "Failed building vtunnel")
+
+	tarballPath := filepath.Join(os.TempDir(), wslTarballName)
+	s, err := os.Stat(tarballPath)
+	// Only download the tarball if
+	// 1) it doesn't exist
+	// 2) the cache is expired > 60 min
+	if errors.Is(err, os.ErrNotExist) {
+		err = downloadFile(tarballPath, wslTarballURL)
+		requireNoErrorf(err, "Failed to download wsl distro tarball %v", wslTarballName)
+	}
+	if s.ModTime().Before(time.Now().Add(time.Hour * -1)) {
+		err = downloadFile(tarballPath, wslTarballURL)
+		requireNoErrorf(err, "Failed to download wsl distro tarball after cache expiry %v", wslTarballName)
+	}
+
+	logrus.Infof("Creating %v wsl distro", wslDistroName)
+	installCmd := cmdExec(
+		tmpDir,
+		"wsl",
+		"--import",
+		wslDistroName,
+		".",
+		tarballPath)
+	err = installCmd.Run()
+	requireNoErrorf(err, "Failed to install distro %v", wslDistroName)
+
+	// It takes a long time to start a new distro,
+	// 20 sec is a long time but that's actually how long
+	// it takes to start a distro without any flakiness
+	timeout := time.Second * 20
```

Comment:
Given that we're just polling in a loop, the timeout length is probably irrelevant — even going to a full minute is fine.